gitclone/checkout_simple_strategy.go

```diff
@@ -0,0 +1,129 @@
+package gitclone
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/bitrise-io/bitrise-init/step"
+	"github.com/bitrise-io/go-utils/command/git"
+)
+
+type checkoutStrategy interface {
+	Validate() error
+	Do(gitCmd git.Git) *step.Error
+}
+
+//
+// checkoutNone
+type checkoutNone struct {
+	ShouldUpdateSubmodules bool
+}
+
+func (c checkoutNone) Validate() error {
+	return nil
+}
+
+func (c checkoutNone) Do(gitCmd git.Git) *step.Error {
+	if c.ShouldUpdateSubmodules {
+		return updateSubmodules(gitCmd)
+	}
+
+	return nil
+}
+
+//
+// checkoutCommit
+type checkoutCommit struct {
+	Commit, branch         string
+	FetchTraits            fetchTraits
+	ShouldUpdateSubmodules bool
```

Comment:
i think update submodules should be detached from the checkouts.

submodule update is independent of the checkout strategy.